### PR TITLE
Fix report missing case

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/thamos/issues/552

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: thamos/cli.py [hunk 1]
```diff
@@ -430,17 +430,20 @@ def advise(
             if json_output:
                 json.dump(result, sys.stdout, indent=2)
             else:
-                stack_info = result.get("report", {}).get("stack_info")
```
Comment (fridex) [inline thamos/cli.py:433]:
```python
                stack_info = (result.get("report") or  {}).get("stack_info")
```

**Comment by saisankargochhayat** [comment]
@pacospace small precommit fix, else looks good :+1:

**Comment by pacospace** [comment]
> @pacospace small precommit fix, else looks good

Not sure how do we solved this `        "long": int if six.PY3 else long,  # noqa: F821` pre-commit keeps failing:

`thamos/swagger_client/api_client.py:53: error: Name 'long' is not defined`

**Comment by saisankargochhayat** [comment]
@pacospace @fridex I think we should exclude pre-commit for generated swagger client and docs because we would have to fix pre-commit every time we regenerate them, maybe add this to all hooks - https://github.com/thoth-station/thamos/blob/53469a7174ad6bcfc5d35fd1b9f7fbf230d8cc06/.pre-commit-config.yaml#L59

**Comment by pacospace** [comment]
> @pacospace @fridex I think we should exclude pre-commit for generated swagger client and docs because we would have to fix pre-commit every time we regenerate them, maybe add this to all hooks -
> 
> https://github.com/thoth-station/thamos/blob/53469a7174ad6bcfc5d35fd1b9f7fbf230d8cc06/.pre-commit-config.yaml#L59

+ 1 for me @harshad16 ?

**Comment by harshad16** [comment]
I think we should change the `long` to `int` as python3 handles it. 
About the exclusion, I don't have a strong opinion on it, however, I feel it good to have `mypy` check on them as well, as it would improve our systems, as we do code regeneration with a pull request (it not automated directly, but run through a script ) so the developer can update it while opening the pull request.

**Comment by pacospace** [comment]
> I think we should change the `long` to `int` as python3 handles it.
> About the exclusion, I don't have a strong opinion on it, however, I feel it good to have `mypy` check on them as well, as it would improve our systems, as we do code regeneration with a pull request (it not automated directly, but run through a script ) so the developer can update it while opening the pull request.

```
thamos/swagger_client/models/python_platforms.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
thamos/swagger_client/models/python_platforms.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
thamos/swagger_client/models/python_package_dependencies.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
thamos/swagger_client/models/python_package_dependencies.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
thamos/swagger_client/models/python_package_indexes.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
thamos/swagger_client/models/python_package_indexes.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
thamos/swagger_client/models/kebechet_webhook_input.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
thamos/swagger_client/models/kebechet_webhook_input.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
thamos/swagger_client/api_client.py:53: error: Name 'long' is not defined
Found 9 errors in 5 files (checked 65 source files)
```

this errors will reappear every time we recreate the client

**Comment by saisankargochhayat** [comment]
> > I think we should change the `long` to `int` as python3 handles it.
> > About the exclusion, I don't have a strong opinion on it, however, I feel it good to have `mypy` check on them as well, as it would improve our systems, as we do code regeneration with a pull request (it not automated directly, but run through a script ) so the developer can update it while opening the pull request.
> 
> ```
> thamos/swagger_client/models/python_platforms.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> thamos/swagger_client/models/python_platforms.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> thamos/swagger_client/models/python_package_dependencies.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> thamos/swagger_client/models/python_package_dependencies.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> thamos/swagger_client/models/python_package_indexes.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> thamos/swagger_client/models/python_package_indexes.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> thamos/swagger_client/models/kebechet_webhook_input.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> thamos/swagger_client/models/kebechet_webhook_input.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> thamos/swagger_client/api_client.py:53: error: Name 'long' is not defined
> Found 9 errors in 5 files (checked 65 source files)
> ```
> 
> These errors will reappear every time we recreate the client though, I adjusted them for this PR

These can be ignored under the swagger client

**Comment by pacospace** [comment]
> > > I think we should change the `long` to `int` as python3 handles it.
> > > About the exclusion, I don't have a strong opinion on it, however, I feel it good to have `mypy` check on them as well, as it would improve our systems, as we do code regeneration with a pull request (it not automated directly, but run through a script ) so the developer can update it while opening the pull request.
> > 
> > 
> > ```
> > thamos/swagger_client/models/python_platforms.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/models/python_platforms.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/models/python_package_dependencies.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/models/python_package_dependencies.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/models/python_package_indexes.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/models/python_package_indexes.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/models/kebechet_webhook_input.py:31: error: Need type annotation for 'swagger_types' (hint: "swagger_types: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/models/kebechet_webhook_input.py:34: error: Need type annotation for 'attribute_map' (hint: "attribute_map: Dict[<type>, <type>] = ...")
> > thamos/swagger_client/api_client.py:53: error: Name 'long' is not defined
> > Found 9 errors in 5 files (checked 65 source files)
> > ```
> > 
> > 
> > These errors will reappear every time we recreate the client though, I adjusted them for this PR
> 
> These can be ignored under the swagger client

```

---
repos:
  - repo: git://github.com/Lucas-C/pre-commit-hooks
    rev: v1.1.1
    hooks:
      - id: remove-tabs
        exclude: '^(thamos/swagger_client/|examples/|docs/)'

  - repo: git://github.com/pre-commit/pre-commit-hooks
    rev: v2.0.0
    hooks:
      - id: trailing-whitespace
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-merge-conflict
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: end-of-file-fixer
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-added-large-files
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-byte-order-marker
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-case-conflict
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-docstring-first
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-json
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-symlinks
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: detect-private-key
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-ast
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: debug-statements
        exclude: '^(thamos/swagger_client/|examples/|docs/)'

  - repo: git://github.com/pycqa/pydocstyle.git
    rev: 4.0.1
    hooks:
      - id: pydocstyle
        exclude: '^(thamos/swagger_client/|examples/|docs/)'

  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v2.3.0
    hooks:
      - id: check-toml
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: check-yaml
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: end-of-file-fixer
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
      - id: trailing-whitespace
        exclude: '^(thamos/swagger_client/|examples/|docs/)'

  - repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.770
    hooks:
      - id: mypy
        exclude: '^(thamos/swagger_client/|examples/|docs/)'
        args: [--ignore-missing-imports]
        additional_dependencies: [attrs~=19.3]

  - repo: https://github.com/psf/black
    rev: 19.10b0
    hooks:
      - id: black
        exclude: '^(thamos/swagger_client/|examples/|docs/)'

  - repo: https://gitlab.com/PyCQA/flake8
    rev: '3.7.8'
    hooks:
      - id: flake8
        additional_dependencies: ['pep8-naming']
        # Ignore all format-related checks as Black takes care of those.
        args: ['--ignore', 'E2,W5', '--select', 'E,W,F,N', '--max-line-length=120']
        exclude: '^(thamos/swagger_client/|examples/|docs/)'

  - repo: https://github.com/mgedmin/check-manifest
    rev: '0.39'
    hooks:
      - id: check-manifest
        exclude: '^(thamos/swagger_client/|examples/|docs/)'

```
I tested with this but the error for int/long is still present

**Comment by harshad16** [comment]
The re-occurring error arent they appearing from user-api files? we should fix them from the source. 
also if we would like to exclude them, lets exclude them only in mypy, wdyt?
should we take this into a new issue, and discuss there. as if we have fixed  the issues in these it would good to let this get merged in.

**Comment by pacospace** [comment]
> The re-occurring error arent they appearing from user-api files? we should fix them from the source.
> also if we would like to exclude them, lets exclude them only in mypy, wdyt?
> should we take this into a new issue, and discuss there. as if we have fixed the issues in these it would good to let this get merged in.

Yes let's have it meanwhile, so the demo on securtiy recommendation can be finished :)

**Comment by pacospace** [comment]
> The re-occurring error arent they appearing from user-api files? we should fix them from the source.
> also if we would like to exclude them, lets exclude them only in mypy, wdyt?
> should we take this into a new issue, and discuss there. as if we have fixed the issues in these it would good to let this get merged in.

can you please open the issue in User-API @harshad16 ?

**Review by fridex — APPROVED** [review]
lgtm :+1:

**Pull request merged**